x/rollup/keeper/msg_server.go

```diff
@@ -83,3 +86,70 @@ func (k *Keeper) InitiateWithdrawal(
 
 	return &types.MsgInitiateWithdrawalResponse{}, nil
 }
+
+func (k *Keeper) InitiateFeeWithdrawal(
+	goCtx context.Context,
+	_ *types.MsgInitiateFeeWithdrawal,
+) (*types.MsgInitiateFeeWithdrawalResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// TODO: make minWithdrawalAmount and l1recipientAddr configurable once a genesis state is added
+	const (
+		minWithdrawalAmount   = 100
+		l1recipientAddr       = "0x63d93aC6FA6B4021527e967ac3Eb29F2B3E52B96"
+		feeWithdrawalGasLimit = 400_000
+	)
+
+	feeCollectorAddr := k.accountkeeper.GetModuleAddress(authtypes.FeeCollectorName)
+	if feeCollectorAddr == nil {
+		ctx.Logger().Error("Failed to get fee collector address")
+		return nil, types.WrapError(types.ErrInitiateFeeWithdrawal, "failed to get fee collector address")
+	}
+
+	feeCollectorBalance := k.bankkeeper.GetBalance(ctx, feeCollectorAddr, types.WEI)
+	if feeCollectorBalance.Amount.LT(math.NewInt(minWithdrawalAmount)) {
+		ctx.Logger().Error("Fee collector balance is below the minimum withdrawal amount", "balance", feeCollectorBalance.String())
+		return nil, types.WrapError(
+			types.ErrInitiateFeeWithdrawal,
+			"fee collector balance is below the minimum withdrawal amount: %v", feeCollectorBalance.String(),
+		)
+	}
+
+	ctx.Logger().Debug("Withdrawing L2 fees", "amount", feeCollectorBalance.String(), "recipient", l1recipientAddr)
+
+	// Burn the withdrawn fees from the fee collector account on L2
+	fees := sdk.NewCoins(feeCollectorBalance)
+	if err := k.bankkeeper.SendCoinsFromModuleToModule(ctx, authtypes.FeeCollectorName, types.ModuleName, fees); err != nil {
```

Comment:
Is it necesssary to send them to the rollup module first?

Reply:
Not necessarily - the alternative here would be to have the app enable burn permissions to the `FeeCollector` account, however I'm not entirely sure if that could open the door for any additional unwanted behavior.

Since we already require the user to enable burn permissions for the `x/rollup` module for user withdrawals, I decided to just forward the fees to the rollup module first before burning the fees.

Lmk what your thoughts on enabling burn permissions for the `FeeCollector` module account are and if you think that would be a better approach.

Reply:
This sounds reasonable. Mind putting that in a comment?